spec/client_spec.rb

```diff
@@ -659,6 +659,12 @@ def loop_times(times)
     expect(resp['error']).to include 'invalid SQS url'
   end
 
+  it 'check_sns with an invalid queue url should fail' do
+    resp = @client.check_sns('key', 'secret', 'arn:aws:sns:us-east-1:123456789012:sns-topic')
+    expect(resp['status']).to eq 'error'
+    expect(resp['error']).to include 'publishing the message failed.'
```

Comment:
I'm not sure what the message from the API is, but this looks like a 5xx error was returned, when the test implies a validation issue (4xx)

Reply:
The endpoint actually returns a 200, and the response contains a dictionary with status either `ok` or `error` and then a few additional properties explaining what the error is.